pkg/config/lima_config_applier.go

```diff
@@ -188,3 +198,46 @@ func hasUserModeEmulationInstallationScript(limaCfg *limayaml.LimaYAML) (int, bo
 
 	return scriptIdx, hasCrossArchToolInstallationScript
 }
+
+func toggleSoci(limaCfg *limayaml.LimaYAML, enabled bool) {
+	idx, hasScript := hasSociInstallationScript(limaCfg)
+	if !hasScript && enabled {
+		limaCfg.Provision = append(limaCfg.Provision, limayaml.Provision{
+			Mode: "system",
+			Script: fmt.Sprintf(`%s
+if [ ! -f /usr/local/bin/soci ]; then
+    #download soci
+    curl -OL "https://github.com/awslabs/soci-snapshotter/releases/download/v0.1.0/soci-snapshotter-0.1.0-linux-arm64.tar.gz"
+    #move to usr/local/bin
+    tar -C /usr/local/bin -xvf soci-snapshotter-0.1.0-linux-arm64.tar.gz ./soci ./soci-snapshotter-grpc
+    #changing containerd config
+    export config=etc/containerd/config.toml
+    //copy config to soci-config
+    echo "    [proxy_plugins.soci]
+        type = \"snapshot\"
+        address = \"/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock\" " >> $config
+sudo systemctl restart containerd.service
+sudo soci-snapshotter-grpc &> ~/soci-snapshotter-logs &
+fi
+`, sociInstallationProvisioningScriptHeader),
+		})
+	} else if hasScript && !enabled {
+		if len(limaCfg.Provision) > 0 {
+			limaCfg.Provision = append(limaCfg.Provision[:idx], limaCfg.Provision[idx+1:]...)
+		}
+	}
+}
+
+func hasSociInstallationScript(limaCfg *limayaml.LimaYAML) (int, bool) {
+	hasSociInstallationScript := false
+	var scriptIdx int
+	for idx, prov := range limaCfg.Provision {
+		trimmed := strings.Trim(prov.Script, " ")
+		if !hasSociInstallationScript && strings.HasPrefix(trimmed, sociInstallationProvisioningScriptHeader) {
+			hasSociInstallationScript = true
+			scriptIdx = idx
+		}
```

Comment:
how about an early break of the loop?

```suggestion
		if strings.HasPrefix(trimmed, sociInstallationProvisioningScriptHeader) {
			hasSociInstallationScript = true
			scriptIdx = idx
			break
		}
```